test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -748,6 +750,8 @@ void UcxConnection::stream_recv_callback(void *request, ucs_status_t status,
     ucx_request *r      = reinterpret_cast<ucx_request*>(request);
     UcxConnection *conn = r->conn;
 
+    // need to check connection status first since it's already can be failed or
+    // disconnecting
```

Comment:
can we check _connected field?

Reply:
no, because it's always false here, `conn->established();` will be called at line 757